File: mobile/lib/models/location_validation_attempt.dart
class LocationValidationAttempt {
  final double latitude;
  final double longitude;
  final double distanceInMeters;
  final double allowedDistanceInMeters;
  final DateTime dateTime;
  final bool validated;

  LocationValidationAttempt({
    required this.latitude,
    required this.longitude,
    required this.distanceInMeters,
    required this.allowedDistanceInMeters,
    required this.dateTime,
    required this.validated,
  });

  Map<String, dynamic> toJson() {
    return {
      'latitude': latitude,
      'longitude': longitude,
      'distanceInMeters': distanceInMeters,
      'allowedDistanceInMeters': allowedDistanceInMeters,
      'dateTime': dateTime.toIso8601String(),
      'validated': validated,
    };
  }
}

File: backend/src/main/java/br/unitins/facelocus/model/LocationValidationAttempt.java
package br.unitins.facelocus.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@EqualsAndHashCode(of = "id", callSuper = false)
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class LocationValidationAttempt {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private String latitude;
    private String longitude;
    private Double distanceInMeters;
    private Double allowedDistanceInMeters;
    private LocalDateTime dateTime;
    private boolean validated;

    @ManyToOne
    private AttendanceRecord attendanceRecord;
}